### DISCUSSION TITLE
Mergify does not seem to execute

### DISCUSSION THREAD

**Discussion by asbjornu** [Mergifyio/mergify#2189] [Q&A]
In SwedbankPay/developer.swedbankpay.com#858, I've executed `@Mergifyio refresh` without Mergify actually doing anything with the pull request. I'm also missing the [summary and check report mentioned in the documentation](https://docs.mergify.io/getting-started.html#understanding-mergify-check-report). Any ideas what may be wonky here?

**Comment by jd** [comment]
Hi @asbjornu!

Your file is misnamed: it's `.github/.mergify.yml` whereas it should be `.github/mergify.yml`.

Reference: https://docs.mergify.io/getting-started.html#the-configuration-file

**Reply by asbjornu** [reply]
Ah! 🤦🏼‍♂️ Thanks! Now Mergify executes, but somehow [failed the build by cancelling its job](https://github.com/SwedbankPay/developer.swedbankpay.com/pull/858/checks?check_run_id=2086601510) due to "Branch protection settings are not validated anymore". I thought that as per #2184, required checks would be respected automatically?

**Reply by jd** [reply]
The rules are respected and applied first: in that case, they match, so the PR is put in the queue.
Then when Mergify tries to merge the PR it fails because it does not have permission to merge. Therefore, the PR is removed from the queue.

**Reply by asbjornu** [reply]
But why doesn't Mergify have permission to merge?

**Reply by jd** [reply]
I guess that's because you have limited authorized users allowed to merge in your branch protection settings.

**Reply by asbjornu** [reply]
Yes, correct! Adding `mergify` helped! 👍🏼 

<img width="890" alt="image" src="https://user-images.githubusercontent.com/12283/110869781-6c302a80-82cb-11eb-8dc8-851779549df6.png">

Thanks!

**Reply by asbjornu** [reply]
Sorry, but now SwedbankPay/developer.swedbankpay.com#872 failed to merge. Any idea why?

**Reply by sileht** [reply]
Looks like your branch protection is blocking the merge:

https://github.com/SwedbankPay/developer.swedbankpay.com/pull/872/checks?check_run_id=2136276184

![image](https://user-images.githubusercontent.com/200878/111621459-4ed9ef80-87e8-11eb-8b82-9fe3ec9a3f60.png)

**Reply by asbjornu** [reply]
Ah, I did not notice that SwedbankPay/developer.swedbankpay.com#858 was reviewed by @arebra. Is it possible to circumvent this branch protection somehow such that pull requests from Dependabot are reviewed and merged automatically?

**Reply by jd** [reply]
No, you'd need to disable branch protections.

**Reply by AraHaan** [reply]
For some reason my PR that adds the config file (I made changes to add some stuff to it by amending the commit then force pushing) but when I try to test to see if it works it does nothing, why is that?

https://github.com/Elskom/Sdk/pull/191

Sadly no ``check-pending=`` exists yet so I had to manually attempt that by doing:

```yml
- -check-success=WIP
- -check-neutral=WIP
- -check-failure=WIP
- -closed
```

**Reply by jd** [reply]
@AraHaan I'm not sure I understand the question/problem. Can you detail?

**Reply by AraHaan** [reply]
I realized later in the docs that you guys intended for the rules to not run whenever the pr changes the config file in any way. But anyway I would at least like a ``check-pending`` condition added for status checks (when github says it's "In Progress", or "Waiting" in the yellow state). Github calls that as their pending status and they even have a whole thing where you can even search prs with ``status:pending`` for check statuses.

**Reply by jd** [reply]
See https://github.com/Mergifyio/mergify-engine/issues/707

**Reply by AraHaan** [reply]
@jd 
![image](https://user-images.githubusercontent.com/15173749/116038170-1921f500-a637-11eb-9add-b027fd6fae18.png)
![image](https://user-images.githubusercontent.com/15173749/116038187-1fb06c80-a637-11eb-9be2-c442837c3e01.png)

For some reason it does not do the failure case properly on the codacy static code analysis check, however it works when my automerge step does ``check-success`` for some reason. Could this be a bug?

My mergify config: https://github.com/Elskom/Sdk/blob/main/.github/mergify.yml

**Reply by sileht** [reply]
That because `Codacy Static Code Analysis` reports "action_required", not "failure". There is a couple of other states that we didn't handle yet: `cancelled` and `timed_out`.

**Reply by AraHaan** [reply]
You guys plan to handle them eventually? Also does ``skipped`` qualify as ``cancelled`` even if it was automatically skipped?

Also is it hard to implement them so as such they been skipped for now?

**Reply by sileht** [reply]
We have a ticket, but we didn't decide yet if we merge them with failure or create a new attribute for each of them.

**Reply by AraHaan** [reply]
I think the ``action_required``, and ``cancelled`` can be merged with failure, and ``skipped`` goes to it's own.
